server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SieveMailet.java

```diff
@@ -220,4 +312,190 @@ public void post(String url, MimeMessage mail) throws MessagingException {
         }
     }
 
+
+    /**
+     * Delivers a mail to a local mailbox.
+     *
+     * @param mail
+     *            the mail being processed
+     *
+     * @throws MessagingException
+     *             if an error occurs while storing the mail
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        Collection<MailAddress> recipients = mail.getRecipients();
+        Collection<MailAddress> errors = new Vector<MailAddress>();
+
+        MimeMessage message = null;
+        if (deliveryHeader != null || resetReturnPath) {
+            message = mail.getMessage();
+        }
+
+        if (resetReturnPath) {
+            // Set Return-Path and remove all other Return-Path headers from the
+            // message
+            // This only works because there is a placeholder inserted by
+            // MimeMessageWrapper
+            message.setHeader(RFC2822Headers.RETURN_PATH,
+                (mail.getSender() == null ? "<>" : "<" + mail.getSender()
```

Comment:
Yes, I extracted it to DeliveryUtils. It should be better to share it IMO.

First merged wins ;-) seconds rebase and adapt ;-)